backend/services/covidService.js

```diff
@@ -45,5 +48,43 @@ module.exports = {
           alertCreated: alert.dateCreated,
         }))
     },
+
+    async getUnassignedNewEntrants(res) {
+      const { caseLoadId } = res.locals.user.activeCaseLoad
+
+      const context = { ...res.locals, requestHeaders: { 'page-offset': 0, 'page-limit': 3000 } }
+
+      const fourteenDaysAgo = now()
+        .startOf('day')
+        .subtract(14, 'days')
+        .format('YYYY-MM-DDTHH:mm:ss')
+      const movements = await elite2Api.getMovementsInBetween(context, caseLoadId, { fromDateTime: fourteenDaysAgo })
+
+      const offenderAlerts = await elite2Api.getAlerts(context, {
+        agencyId: caseLoadId,
+        offenderNumbers: movements.map(inmate => inmate.offenderNo),
+      })
+
+      const alertsByOffenderNumber = offenderAlerts.reduce((result, alert) => {
+        const bucket = result[alert.offenderNo] || []
+        const newBucket = [...bucket, alert]
+        return { ...result, [alert.offenderNo]: newBucket }
```

Comment:
Is this necessary? The arrays and objects aren't immutable. You could just push the new alert onto the bucket. Likewise with spreading result into a new object.

Reply:
There's obviously object creation but its pretty explicit.
With linting rules it becomes a bit too verbose and fiddly:
```
        const bucket = result[alert.offenderNo] || []
        bucket.push(alert)
        const newResult = result
        newResult[alert.offenderNo] = bucket
        return newResult
```